firmware-binaries/test-cases/switch_calendar_test/src/main.rs

```diff
@@ -0,0 +1,165 @@
+#![no_std]
+#![cfg_attr(not(test), no_main)]
+#![allow(const_item_mutation)]
+#![allow(clippy::empty_loop)]
+#![allow(clippy::approx_constant)]
+
+use bittide_hal::manual_additions::index::IndexTy;
+use bittide_hal::switch_c::ValidEntry;
+use bittide_hal::Index;
+// SPDX-FileCopyrightText: 2025 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+use ufmt::uwrite;
+
+use bittide_hal::hals::switch_c as hal;
+use core::fmt::Write;
+#[cfg(not(test))]
+use riscv_rt::entry;
+
+const INSTANCES: hal::DeviceInstances = unsafe { hal::DeviceInstances::new() };
+
+fn test_result(result: &str) -> ! {
+    let uart = &mut INSTANCES.uart;
+    writeln!(uart, "RESULT: {}", result).unwrap();
```

Comment:
`uwritln!` could help reduce binary size and should be preferred whenever possible